lib/symboldatabase.cpp

```diff
@@ -4911,10 +4949,12 @@ const Function* Scope::findFunction(const Token *tok, bool requireConst) const
 
 //---------------------------------------------------------------------------
 
-const Function* SymbolDatabase::findFunction(const Token *tok) const
+const Function* SymbolDatabase::findFunction(const Token *tok, const Scope* currScope) const
```

Comment:
scope is only set on tokens after variables have been processed. I'm not sure why. The scopes are already created and know their contents so if I supply the scope (which I have) here then everything works at this early stage. If nullptr scope is supplied (the default) then it reverts to the old behavior.

Reply:
You mean to modify tok->scope here? That wouldn't be possible because ```tok``` is const which I think is the right design decision for a function called findFunction. You wouldn't expect that to modify the input  token. It would be nice if at some point prior to calling findFunction all the tokens scopes were configured correctly. I'm not sure what that would involve or whether it would be worthwhile.